src/pods/interface/fs_interface.rs

```
@@ -270,43 +276,43 @@ impl FsInterface {
+                            .network_interface
+                            .callbacks
+                            .resolve(Callback::Pull(id), false)
+                    }
+                },
+                arbo.get_inode(id)?.clone(),
+            )
         };
+
         let status = self.disk.write_file(path, &binary, 0).is_ok();
-        self.network_interface
+        let _ = self.network_interface
             .callbacks
-            .resolve(Callback::Pull(id), status)?;
+            .resolve(Callback::Pull(id), status);
         if status {
             let mut hosts;
-            {
-                let inode = arbo.get_inode(id)?;
-                if let FsEntry::File(hosts_source) = &inode.entry {
-                    hosts = hosts_source.clone();
-                    let self_addr = self.network_interface.self_addr.clone();
-                    let idx = hosts.partition_point(|x| x <= &self_addr);
-                    hosts.insert(idx, self_addr);
-                } else {
-                    return Err(io::ErrorKind::InvalidInput.into());
-                }
+            if let FsEntry::File(hosts_source) = &inode.entry {
+                hosts = hosts_source.clone();
+                let self_addr = self.network_interface.self_addr.clone();
+                let idx = hosts.partition_point(|x| x <= &self_addr);
+                hosts.insert(idx, self_addr);
+            } else {
+                return Err(io::ErrorKind::InvalidInput.into());
             }
-            arbo.set_inode_hosts(id, hosts)?;
-            let inode = arbo.get_inode(id)?;
-            self.network_interface.update_remote_hosts(inode)
+            Arbo::write_lock(&self.arbo, "recept_binary")
+                .expect("recept_binary: can't write lock arbo")
+                .set_inode_hosts(id, hosts)?;
+            self.network_interface.update_remote_hosts(&inode)
         } else {
-            Ok(())
+            Ok(()) // REVIEW - should be error ?
```

is there a reason to ignore the error ?
It seems to me like it could be simply dealt with a `?` on line 296 instead of `is_ok`